app/controllers/admin/api/api_docs_services_controller.rb

```diff
@@ -132,8 +133,20 @@ def api_docs_params(*extra_params)
     params.require(:api_docs_service).permit(*permit_params)
   end
 
+  def api_docs_services
+    collection = current_account.api_docs_services.accessible
+    return collection unless current_user
+    collection.permitted_for(current_user)
+  end
+
   def find_api_docs_service
-    @api_docs_service = current_account.api_docs_services.find(params[:id])
+    @api_docs_service = api_docs_services.find(params[:id])
   end
 
+  def find_service
+    service_id = api_docs_params[:service_id]
+    service_id.blank? || current_user.blank? || current_user.accessible_services.find(service_id)
+  rescue ActiveRecord::RecordNotFound
+    render_error('Service not found', status: :unprocessable_entity)
```

Comment:
`not_found` or `unprocessable_entity`? I'd prefer the first one as the error clearly mentioned it.
You try to find an object that belongs to a resource that does not exist.

Applying unprocessable_entity would be for example, trying to query an object that exists but in a state that cannot be processed.

Reply:
Actually, we have 3 options here:
- not found
- unprocessable entity
- forbidden

In reality, what happens here is that the service is either _forbidden_ or _not found_. The reason why I didn't go with _not found_ is because the resource targeted is the api doc spec (not the service) and the api doc spec may exist. It would be weird to fetch an existing api doc spec, then try to modify the parameter `service_id` of the spec to a value that is not acceptable and receive a 404 as response, but a 200 when the value if acceptable for the parameter. The target resource is not the one missing and it reveals too much to the user about the dependencies between objects.

Finally, I didn't go with _forbidden_ because it's considered a good practice not to distinguish between "forbidden" and "not found" when a resource exists but it's forbidden to the requestor. Otherwise, we can open a breach for exploring existing object ids in the system.